### PR TITLE
MRG: Organise NIRS specific preprocessing

### PR DESCRIPTION
**Pull request by rob-luke**
#### What does this implement/fix?
Address issues raised  by @agramfort in #7035

### CODE REVIEW

**Comment by rob-luke** [comment]
@agramfort please let me know if I missed anything.

File: mne/io/nirx/tests/test_nirx.py [hunk 1]
```diff
@@ -13,8 +13,8 @@
 from mne.io.tests.test_raw import _test_raw_reader
 from mne.transforms import apply_trans, _get_trans
 from mne.utils import run_tests_if_main
-from mne.preprocessing._beer_lambert_law import _probe_distances,\
-    short_channels
+from mne.preprocessing.nirs import nirs_source_detector_distances,\
+    nirs_short_channels
```
Comment (agramfort) [inline mne/io/nirx/tests/test_nirx.py]: if you put functions in a nirs submodule no need to replicate nirs in the function name.

you could do

```
from mne.preprocessing import nirs
nirs.source_detector_distances(...)
```

File: mne/preprocessing/_beer_lambert_law.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 from ..io.pick import _picks_to_idx
 from ..io.constants import FIFF
 from ..utils import _validate_type
+from mne.preprocessing.nirs import nirs_source_detector_distances
```
Comment (agramfort) [inline mne/preprocessing/_beer_lambert_law.py]: relative import please

Comment (agramfort) [reply mne/preprocessing/_beer_lambert_law.py]: move also this file to nirs submodule

**Comment by larsoner** [comment]
@rob-luke can you please open an `ENH: Agenda for fNIRS processing` issue or so? There are things I'd like to put in / discuss there. For example whether or not [TDDR](https://github.com/frankfishburn/TDDR) is something worth adding. Let's discuss that over in the new issue, though, so we don't hijack this.

(I would create the issue myself, but if you create it then we'll both have edit permissions on the top comment.)

File: doc/python_reference.rst [hunk 3]
```diff
@@ -362,9 +362,22 @@ Projections:
...
+   :no-inherited-members:
+
+.. autosummary::
+   :toctree: generated/
+
    optical_density
    beer_lambert_law
-   short_channels
+   nirs_source_detector_distances
+   nirs_short_channels
```
Comment (larsoner) [inline doc/python_reference.rst]: missed the `nirs_` removals here

Comment (rob-luke) [reply doc/python_reference.rst]: 😉

**Comment by rob-luke** [comment]
@larsoner & @agramfort this is ready for review again. Thanks

File: mne/preprocessing/nirs/__init__.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+"""NIRS specific preprocessing functions."""
```
Comment (agramfort) [inline mne/preprocessing/nirs/__init__.py]: Can you had header to files with your name and license thx

**Comment by larsoner** [comment]
Pushed license+authors, thanks @rob-luke !

**Pull request merged**